transcom/mymove
# Int b 17960 integrate calc display

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a869501)

## Summary

This PR updates the PPM Closeout "Review Documents" workflow to include more information about a PPM shipment and the costs/calculations behind the incentives for it. There is now a "Shipment Info" dropdown that shows general info such as move dates, ZIPs, and mileage. 

Then the "Incentives/Costs" section breaks down the GCC cost, gross incentive, any advances received, and any remaining incentive owed to the customer.

Finally, the "GCC Factors" section includes relevant info that contributes to the value of the "GCC (Government Constructed Cost)", such as haul prices and packing expenses.

### How to test

1. Access http://milmovelocal:3000/sign-in and sign in as a new customer

2. Create a new PPM move with one or more PPM shipments. Take note of certain values when making the shipments (not when making profile info), such as 
- Origin / Destination Zips
- Planned move date
- Estimated weight
- Pro gear self and spouse estimated weight (if applicable)
- Estimated incentive
- Advance amount requested (if applicable)

If making multiple shipments be sure to note these values *for each shipment*. Finish up and sign the form to finish creating the move. Take note of the move code.

3. Login as a SC at http://officelocal:3000/sign-in and click on the new move.
4. Edit the orders to include needed info.

5. Click "Edit Shipment" and put in the estimated SIT weight if applicable (*NOTE: Be sure to put in the estimated SIT weight, start and end date if the customer marked that SIT is expected, if not these fields will be stuck at null in the DB and SIT related expenses will not be taken into account for calculations*).

6. If an advance was requested, click through to the next screen and approve it.

7. Click "Submit Move Details"

8. Go back to the customer with the move and click "Upload PPM documents". Take note of *all* of the values you put in for these fields throughout the whole process, as they will be needed to verify the calculations later. If you indicated things like "Pro gear" be sure to add weights for those things, and I recommend putting in different "actual" values on these screens than what you put for "estimated", this will help you verify that the calculation screen is showing the "actual" value and not the "estimated" one.

9. You can also add other misc. expenses that will show in the review screen. These will also contribute to the incentive totals.

10. After signing for the PPM documents, go back to the SC and access the move under the PPM Closeout tab. Click "Review Documents" on the shipment(s).

11. Before approving any receipts, you should see a Shipment Info dropdown, click and verify that the values look correct.

![Screenshot 2024-02-13 at 3 59 51 PM](https://github.com/transcom/mymove/assets/146959895/8f6b2b2b-311a-4485-8849-3812fc715037)

12. After approving all receipts, before clicking "Confirm", you should see two new sections ("Incentives" and "GCC Factors"), check to make sure that these at least make sense (i.e. GCC should always be higher or equal to total incentive).

![Screenshot 2024-02-13 at 4 00 00 PM](https://github.com/transcom/mymove/assets/146959895/029e6a81-9442-4b47-8607-d6835b7a37a4)

![Screenshot 2024-02-13 at 4 00 11 PM](https://github.com/transcom/mymove/assets/146959895/06310af0-5fd9-432e-895c-c929fe164686)

You should also see the correct breakdown for all documents/tickets:
![Screenshot 2024-02-13 at 4 00 32 PM](https://github.com/transcom/mymove/assets/146959895/88def594-ce1b-44e7-b565-251f3b32c8cb)

13. Test that fields such as GCC update when you go back to the "Trip" weight receipt and update the allowable weight. GCC uses the max allowable weight to calculate the max allowable incentive amount.
for Trip tickets should result in a lower incentive, vice versa for full weight. Rejecting a Trip weight ticket should also lower the amount.

**P.S.:** There is currently a bug in the documents screen where editing the weights (empty/full) in one Trip ticket edits the weights for all Trip tickets on the front end. I.e. when going back and forth between trips you'll need to make sure you put in different weights if you want. This already existed in the app as far as I know and I would consider this out of scope for the story.

## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Frontend


### Backend
